internal/internal_worker.go

```diff
@@ -317,19 +318,26 @@ func newWorkflowTaskWorkerInternal(
 ) *workflowWorker {
 	ensureRequiredParams(&params)
 	poller := newWorkflowTaskPoller(taskHandler, contextManager, service, params)
+	workerType := "WorkflowWorker"
+	metricsHandler := params.MetricsHandler.WithTags(metrics.WorkerTags(workerType))
+	tss := newTrackingSlotSupplier(params.Tuner.GetWorkflowTaskSlotSupplier(), metricsHandler)
 	worker := newBaseWorker(baseWorkerOptions{
-		pollerCount:       params.MaxConcurrentWorkflowTaskQueuePollers,
-		pollerRate:        defaultPollerRate,
-		maxConcurrentTask: params.ConcurrentWorkflowTaskExecutionSize,
-		maxTaskPerSecond:  defaultWorkerTaskExecutionRate,
-		taskWorker:        poller,
-		identity:          params.Identity,
-		workerType:        "WorkflowWorker",
-		stopTimeout:       params.WorkerStopTimeout,
-		fatalErrCb:        params.WorkerFatalErrorCallback,
+		pollerCount:      params.MaxConcurrentWorkflowTaskQueuePollers,
+		pollerRate:       defaultPollerRate,
+		slotSupplier:     tss,
+		maxTaskPerSecond: defaultWorkerTaskExecutionRate,
+		taskWorker:       poller,
+		identity:         params.Identity,
+		workerType:       workerType,
+		stopTimeout:      params.WorkerStopTimeout,
+		fatalErrCb:       params.WorkerFatalErrorCallback,
+		metricsHandler:   metricsHandler,
+		slotReservationData: slotReservationData{
+			taskQueue: params.TaskQueue,
+			logger:    params.Logger,
```

Comment:
Wouldn't you want the logger here tagged? This is also identical in the 4 calls to `newBaseWorker` so I think we can just move it into `newBaseWorker`

Reply:
Moved